nsdb-core/src/main/scala/io/radicalbit/nsdb/post_proc/package.scala

```diff
@@ -159,4 +168,21 @@ package object post_proc {
     )
   }
 
+  def internalAggregationProcessing(bits: Seq[Bit],
+                                    schema: Schema,
+                                    aggregationType: InternalSimpleAggregationType): Bit = {
+    val v                              = schema.value.indexType.asInstanceOf[NumericType[_]]
+    implicit val numeric: Numeric[Any] = v.numeric
+    aggregationType match {
+      case InternalMaxSimpleAggregation(_, _) =>
```

Comment:
i think by using type checking you can save some `unapply` time

Reply:
thank you! Will update